src/plugins/ui_feature/remove_duplicates.ts

```diff
@@ -0,0 +1,167 @@
+      this.dispatch("CLEAR_CELL", { col, row, sheetId });
+    }
+
+    const zonePasted: Zone = {
+      left: zone.left,
+      top: zone.top,
+      right: zone.left,
+      bottom: zone.top,
+    };
+
+    state.paste([zonePasted]);
+
+    const remainingZone = {
+      left: zone.left,
+      top: zone.top - (hasHeader ? 1 : 0),
+      right: zone.right,
+      bottom: zone.top + remainingRows - 1,
+    };
+
+    this.selection.selectZone({
+      cell: { col: remainingZone.left, row: remainingZone.top },
+      zone: remainingZone,
+    });
+
+    const removedRows = zone.bottom - zone.top + 1 - remainingRows;
+    this.notifyRowsRemovedAndRemaining(removedRows, remainingRows);
+  }
+
+  private getUniqueRowsIndex(sheetId: UID, zone: Zone, columnsToAnalyze: HeaderIndex[]): string[] {
+    const uniqueRows: CellValueByRow = {};
+
+    for (let rowIndex = zone.top; rowIndex <= zone.bottom; rowIndex++) {
+      // Ensures that the columns to be analyzed are in the the zone
+      const colToAnalyze = columnsToAnalyze.filter(
+        (colIndex) => zone.left <= colIndex && colIndex <= zone.right
+      );
+
+      const cellsValueToAnalyze = colToAnalyze.map((colIndex) => {
+        return this.getters.getEvaluatedCell({
+          sheetId,
+          col: colIndex,
+          row: rowIndex,
+        }).value;
+      });
+
+      const isRowUnique = !Object.values(uniqueRows).some((uniqueRow) => {
+        return deepEquals(uniqueRow, cellsValueToAnalyze);
+      });
+
+      if (isRowUnique) {
+        uniqueRows[rowIndex] = cellsValueToAnalyze;
+      }
+    }
+    return Object.keys(uniqueRows);
+  }
+
+  private notifyRowsRemovedAndRemaining(removedRows: number, remainingRows: number) {
+    this.ui.notifyUI({
+      type: "ERROR",
+      text: _lt(
+        "%s duplicate rows found and removed.\n%s unique rows remain.",
+        removedRows.toString(),
+        remainingRows.toString()
+      ),
+    });
+  }
+
+  private checkSingleRangeSelected(): CommandResult {
+    const zones = this.getters.getSelectedZones();
+    if (zones.length !== 1) {
+      return CommandResult.MoreThanOneRangeSelected;
+    }
+    return CommandResult.Success;
+  }
+
+  private checkNoMergingZone(): CommandResult {
+    const sheetId = this.getters.getActiveSheetId();
+    const zone = this.getters.getSelectedZone();
+    const mergesInZone = this.getters.getMergesInZone(sheetId, zone);
+    if (mergesInZone.length > 0) {
+      return CommandResult.WillRemoveExistingMerge;
+    }
+    return CommandResult.Success;
+  }
+
+  private checkRangeContainsValues(): CommandResult {
+    const sheetId = this.getters.getActiveSheetId();
+    const zones = this.getters.getSelectedZones();
+    const evaluatedCells = this.getters.getEvaluatedCellsInZone(sheetId, zones[0]);
+    if (evaluatedCells.every((evaluatedCel) => evaluatedCel.type === "empty")) {
+      return CommandResult.NoValueSelected;
+    }
+    return CommandResult.Success;
+  }
```

Comment:
I don't think there is a functional benefit to having this behavior. I think that if GSheet works like that, it's because of a side effect.
Especially as you say, the error message is "Please select a range of cells containing values".